### ISSUE TITLE
[WIP] Page loadtimes can take 15 times longer under specific usage conditions

### ISSUE THREAD

**Issue by unlikelyzero** [ansible/awx#4946]
##### ISSUE TYPE
 - Bug Report

##### SUMMARY
Note: this ticket is WIP and needs more information and repro steps.

I experienced extremely slow page load performance while using the container groups feature on a recent build. After some time (maybe an hour) of testing the container groups feature, the page load performance was so poor that the session became unusable. 

Notes:
- At first I thought it was related to real-time updates from the Job Results detail page, but the performance was noticeably worse on all parts of the app. 
- At the time, I was exercising different failure modes of JTs launched by a failed awx-runner container with the default admin user and system auditor.
- The slowness occured on a Firefox ESR browser.
- To characterize the behavior, I measured the following loadtimes on https://<URL>/#/jobs/playbook/30?job_search=page_size:20;order_by:-finished;not__launch_type:sync

```
Cache Disabled Chrome: 1.1s
Cache Disabled Firefox: 15.5s
Cache Enabled Chrome: 0.36s
Cache Enabled Firefox: 8.47s
```

These timings show that the root cause was not specific to any cached or static assets, but because chrome was launched only to take secondary measurements, it's uncertain whether the behavior is specific to Firefox.

TBD

- Is this a memory leak present which only manifests itself after an hour
- Is this an issue specific to the container group feature
- Is this an issue specific to isolated jobs / container groups while viewing many JT runs

##### ENVIRONMENT
* AWX version: 7.0.0
* Operating System: macOS
* Web Browser: Firefox 68.1.0esr

##### STEPS TO REPRODUCE

WIP

##### ADDITIONAL INFORMATION

WIP

**Comment by ryanpetrello** [comment]
@unlikelyzero do you have any before/after numbers to quantity what exactly regressed here?  Is this really `priority:high` ?

(what do we need to get this out of WIP and quantify what's wrong and whether it belongs in 3.6.0?)

**Comment by unlikelyzero** [comment]
@ryanpetrello yes, the numbers are in the issue. It was taking 15x as long to load that page in Firefox.

I'll need time to further isolate the issue to get it out of WIP but I wouldn't take it out of 3.6.0. It's assigned to me.

For anyone who is interested in repro, I believe there is some sort of performance issue around isolated instance / container groups and the job results detail page.

**Comment by ryanpetrello** [comment]
Do you have any more details on this @unlikelyzero?  How readily reproducible is it?

**Comment by unlikelyzero** [comment]
No update at this time

**Comment by jakemcdermott** [comment]
I attempted to reproduce the slow behavior using ffox esr 68 last Wednesday, but didn't see anything out of the ordinary in terms of page load speed or performance.

![Screenshot from 2019-10-16 16-22-21](https://user-images.githubusercontent.com/9753817/67422281-207ac500-f5a0-11e9-82dc-f88507bacc88.png)

I've communicated my findings to @unlikelyzero, to my knowledge he still wants to spend time on this issue at some point.

That said, I don't intend to take another look at this without more information or specific details on how to actually reproduce it.
